src/render/src/macros/mod.rs

```diff
@@ -31,7 +31,99 @@ macro_rules! gfx_format {
 }
 
 
-/// Defines vertex, constant and pipeline formats in one block
+/// Defines vertex, constant and pipeline formats in one block.
+///
+/// # Example
+///
+/// ```{.rust}
+/// #[macro_use]
+/// extern crate gfx;
+///
+/// gfx_defines! {
+///     vertex Vertex {
+///         pos: [f32; 4] = "a_Pos",
+///         tex_coord: [f32; 2] = "a_TexCoord",
+///     }
+///     
+///     constant Locals {
+///         transform: [[f32; 4]; 4] = "u_Transform",
+///     }
+///
+///     pipeline pipe {
+///         vbuf: gfx::VertexBuffer<Vertex> = (),
+///         transform: gfx::Global<[[f32; 4]; 4]> = "u_Transform",
```

Comment:
maybe add a comment to say that this is for compatibility when constant buffers are not supported ?